examples/api/lib/painting/axis_direction/axis_direction.0.dart

```diff
@@ -0,0 +1,208 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for [AxisDirection]s.
+
+import 'package:flutter/material.dart';
+
+void main() => runApp(const ExampleApp());
+
+class ExampleApp extends StatelessWidget {
+  const ExampleApp({super.key});
+
+  static const String _title = 'Flutter Code Sample';
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      title: _title,
+      home: MyWidget(),
+    );
+  }
+}
+
+class MyWidget extends StatefulWidget {
+  const MyWidget({ super.key });
+
+  @override
+  State<MyWidget> createState() => _MyWidgetState();
+}
+
+class _MyWidgetState extends State<MyWidget> {
+  final List<String> alphabet = <String>[
+    'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O',
+    'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z',
+  ];
+  final Widget spacer = const SizedBox.square(dimension: 10);
+  Axis axis = Axis.vertical;
+  bool reverse = false;
+
+  AxisDirection _getAxisDirection() {
+    switch (axis) {
+      case Axis.vertical:
+        return reverse ? AxisDirection.up : AxisDirection.down;
+      case Axis.horizontal:
+        return reverse ? AxisDirection.left : AxisDirection.right;
+    }
+  }
+
+  Widget _getArrows() {
+    final Widget arrow;
+    switch(_getAxisDirection()) {
+      case AxisDirection.up:
+        arrow = const Icon(Icons.arrow_upward_rounded);
+      case AxisDirection.down:
+        arrow = const Icon(Icons.arrow_downward_rounded);
+      case AxisDirection.left:
+        arrow = const Icon(Icons.arrow_back_rounded);
+      case AxisDirection.right:
+        arrow = const Icon(Icons.arrow_forward_rounded);
+    }
+
+    switch(axis) {
+      case Axis.vertical:
+        return Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[ arrow, arrow ]
+        );
+      case Axis.horizontal:
+        return Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[ arrow, arrow ]
+        );
+    }
+  }
+
+  void _onAxisDirectionChanged(AxisDirection? axisDirection) {
+    switch(axisDirection) {
+      case AxisDirection.up:
+        reverse = true;
+        axis = Axis.vertical;
+      case AxisDirection.down:
+        reverse = false;
+        axis = Axis.vertical;
+      case AxisDirection.left:
+        reverse = true;
+        axis = Axis.horizontal;
+      case AxisDirection.right:
+        reverse = false;
+        axis = Axis.horizontal;
+      case null:
+    }
+    setState((){
+      // Respond to change in axis direction.
```

Comment:
Empty setStates are kinda a smell. I would maybe refactor this and do a `if (axisDirection == null) return;` at the beginning and then wrap the entire switch-case (without the null case) in the setState. Or you could have the switch case assign the values to temp variables and then check down here if they are different from the property and then assign them to the property in the setState.